tests/e2etest-azure-platform/main.k

```diff
@@ -0,0 +1,151 @@
+                        }
+                    }
+                },
+
+                # 2. Database claim (MySQL)
+                platformazurev1alpha1.XSQLInstance{
+                    metadata = {
+                        name: "platform-ref-azure-cluster-db-mysql"
+                    }
+                    spec = {
+                        compositionSelector = {
+                            matchLabels = {
+                                dbengine: "mysql"
+                            }
+                        }
+                        parameters = {
+                            region: "westeurope"
+                            storageGB: 20  # Minimum 20 GB for MySQL
+                            version: "8.0"
+                            passwordSecretRef = {
+                                namespace: "default"
+                                name: "mysqlsecret"
+                                key: "password"
+                            }
+                            networkRef = {
+                                id: "platform-ref-azure-cluster"
+                            }
+                        }
+                        writeConnectionSecretToRef = {
+                            name: "platform-ref-azure-cluster-db-conn-mysql"
+                            namespace: "default"
+                        }
+                    }
+                },
+
+                # 3. App claim (Ghost blog application)
+                platformv1alpha1.XApp{
+                    metadata = {
+                        name: "platform-ref-azure-cluster-ghost"
+                        labels = {
+                            "platform.upbound.io/deletion-ordering": "enabled"
+                        }
+                    }
+                    spec = {
+                        parameters = {
+                            providerConfigName: "platform-ref-azure-cluster"
+                            helm = {
+                                wait: False
+                            }
+                            passwordSecretRef = {
+                                namespace: "default"
+                                name: "platform-ref-azure-cluster-db-conn-mysql"
+                            }
+                        }
+                        writeConnectionSecretToRef = {
+                            name: "platform-ref-azure-cluster-app-conn"
+                            namespace: "default"
+                        }
+                    }
+                }
+            ]
+            extraResources: [
+                # Database password secret
+                {
+                    apiVersion: "v1"
+                    kind: "Secret"
+                    metadata = {
+                        name: "mysqlsecret"
+                        namespace: "default"
+                    }
+                    type: "Opaque"
+                    data: {
+                        password: "dXBiMHVuZHIwY2s1ITMxMzM3"
```

Comment:
Can we use the random geneated one? Example can be found in configuration-app

Reply:
Would you be so kind to point to a specific implementation line? :) 

Overall, MySQL FlexibleServer is missing autogenerate password feature in provider-azure, when we add it to provider, it will be easier to switch